projects/extension/RELEASE_NOTES.md

```diff
@@ -1,7 +1,75 @@
+- Added cache for secrets 20809c16745540bd8bc21a546f0d0b7ec912549e
+- Allowed drop_vectorizer to optionally drop target and view ec53befe9151a2d0091de53ace068b8ea2f12573
+- Added an api_key_name parameter to allow functions to remain immutable while 
+  getting a secret. This avoids having to use ai.reveal_secret() which is stable 
+  and not immutable. This allows for more efficient queries when getting a secret 
+  that is not the one with a default query. 59b86d66f92840eed49f80e9ebdcf4f0c60475bd
+- Made reveal_secrets stable f2e0e1489f2ac30f824db7ff137e1252463bddb1
+- Added an event trigger to detect when a source, queue, or target table associated 
+  with a vectorizer is dropped. The event trigger calls ai.drop_vectorizer to 
+  clean up the mess. a01e6208e81942b289970feebfc96bafb95c3fcc
+- Allowed SQL to be gated behind feature flags. This commit added support for building 
+  and shipping prerelease SQL code gated behind feature flags in extension versions 
+  that include a prerelease tag. Prerelease SQL code is omitted entirely from 
+  extension versions that do not include a prerelease tag. For details, see 
+  DEVELOPMENT.md d2bcbfaa83f424d9b8d6894d4d206be8f84ab8d6
+- Added tests to check that extension upgrades work d2bcbfaa83f424d9b8d6894d4d206be8f84ab8d6
+
+### Fixes
+
+- Made ai.secret_permissions dump/restore properly. Two rows are inserted into 
+  ai.secret_permissions on extension creation. If these rows are dumped, then on 
+  restore the extension creation inserts them AND the table restoration tries to insert them.
+  This causes a constraint violation and prevents the table from restoring properly.
+  This commit added a filter to prevent the two rows from being dumped. 39d61db97e85f61441dbe2eafa2bee209bc797fd
+- Prevent vectorizer status view from failing if missing privileges to one or more
+  vectorizer queue tables 44ea1cb0f92b294284ae252fd179191d83145d5c
+- Handle dropped columns when creating vectorizers 814f0ba5a27d69f839c7c8232b118a7a4d0e6772
+- Avoid inserting duplicates into ai._secret_permissions. This fixes an issue that
+  would cause upgrade to fail. ec2363a9f55cc25ce1295526c9f90d9446edd97b
+
+### Breaking changes
+
+- Previously, the vectorizer_queue_pending function would return an exact count which could be very
+  slow for queues with a large number of rows. Now, by default, we limit the count to 10000 by
+  default. An `exact_count` parameter was added. If true, the original behavior is used. 
+  c11db9c2d7fb8346f28f4de17bf3706e9d1620d4
+- If a vectorizer has no queue table, or the user does not have select privileges on the queue table
+  we will now return null for the pending_items column in the vectorizer_status view. 
+  f17d1b908df9fd7072b5554de7dc162102a5611b
+
+### Deprecations
+
+- Versions `0.1.0`, `0.2.0`, and `0.3.0` are deprecated and will be removed in a future release.
+  You must `DROP EXTENSION ai` and then `CREATE EXTENSION ai VERSION '0.4.1' CASCADE`.
```

Comment:
I am aware. My point was that it's not necessarily clear that the advice in the second sentence only applies to the first sentence. My suggestion would be:

```suggestion
  To upgrade from a deprecated version, you must `DROP EXTENSION ai` and then `CREATE EXTENSION ai VERSION '0.4.1' CASCADE`.
```

Reply:
Ahhh. okay. I gotcha.